### ISSUE TITLE
Segmentation fault when calling nest.Simulate() after importing mpi4py

### ISSUE THREAD

**Issue by cristianoalessandro** [nest/nest-simulator#1703]
**Describe the bug**
Segmentation fault if I call nest.Simulate after importing mpi4py

**To Reproduce**
```
$ python
>>>from mpi4py import MPI
>>>import nest
>>>nest.Simulate(10)

Jul 29 12:48:38 NodeManager::prepare_nodes [Info]: 
    Preparing 0 nodes for simulation.

Jul 29 12:48:38 MUSICManager::enter_runtime [Info]: 
    Entering MUSIC runtime with tick = 0.1 ms
[etna:07475] *** Process received signal ***
[etna:07475] Signal: Segmentation fault (11)
[etna:07475] Signal code: Address not mapped (1)
[etna:07475] Failing at address: 0x59
[etna:07475] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20)[0x7f5a8c5eff20]
[etna:07475] [ 1] /home/xis/HBP/MUSIC-1.1.16/lib/libmusic.so.1(_ZN5MUSIC5Setup19maybePostponedSetupEv+0x0)[0x7f5a52e4ea70]
[etna:07475] [ 2] /home/xis/HBP/MUSIC-1.1.16/lib/libmusic.so.1(_ZN5MUSIC7RuntimeC1EPNS_5SetupEd+0x1aa)[0x7f5a52e475ca]
[etna:07475] [ 3] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libnestkernel.so(_ZN4nest12MUSICManager13enter_runtimeEd+0xbe7)[0x7f5a53bed997]
[etna:07475] [ 4] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libnestkernel.so(_ZN4nest17SimulationManager7prepareEv+0x15b)[0x7f5a53c03edb]
[etna:07475] [ 5] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libnestkernel.so(_ZN4nest17SimulationManager8simulateERKNS_4TimeE+0x11)[0x7f5a53c0c9c1]
[etna:07475] [ 6] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libnestkernel.so(_ZN4nest8simulateERKd+0xc2)[0x7f5a53bf19a2]
[etna:07475] [ 7] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libnestkernel.so(_ZNK4nest10NestModule16SimulateFunction7executeEP14SLIInterpreter+0x43)[0x7f5a53bbda83]
[etna:07475] [ 8] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libsli.so(_ZN13FunctionDatum7executeEP14SLIInterpreter+0x43)[0x7f5a5360f053]
[etna:07475] [ 9] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libsli.so(_ZN14SLIInterpreter8execute_Em+0x222)[0x7f5a5360cd32]
[etna:07475] [10] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/../../../libsli.so(_ZN14SLIInterpreter7executeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x15e)[0x7f5a5360d2ae]
[etna:07475] [11] /home/xis/HBP/nest-2.18.0/lib/python3.6/site-packages/nest/pynestkernel.so(+0x37d54)[0x7f5a552ffd54]
[etna:07475] [12] /home/xis/.pyenv/versions/bsb/bin/python[0x50a35c]
[etna:07475] [13] /home/xis/.pyenv/versions/bsb/bin/python(_PyEval_EvalFrameDefault+0x444)[0x50bfb4]
[etna:07475] [14] /home/xis/.pyenv/versions/bsb/bin/python[0x509758]
[etna:07475] [15] /home/xis/.pyenv/versions/bsb/bin/python[0x50a48d]
[etna:07475] [16] /home/xis/.pyenv/versions/bsb/bin/python(_PyEval_EvalFrameDefault+0x444)[0x50bfb4]
[etna:07475] [17] /home/xis/.pyenv/versions/bsb/bin/python[0x507d64]
[etna:07475] [18] /home/xis/.pyenv/versions/bsb/bin/python[0x588d41]
[etna:07475] [19] /home/xis/.pyenv/versions/bsb/bin/python(PyObject_Call+0x3e)[0x59fc4e]
[etna:07475] [20] /home/xis/.pyenv/versions/bsb/bin/python(_PyEval_EvalFrameDefault+0x17e6)[0x50d356]
[etna:07475] [21] /home/xis/.pyenv/versions/bsb/bin/python[0x507d64]
[etna:07475] [22] /home/xis/.pyenv/versions/bsb/bin/python[0x509a90]
[etna:07475] [23] /home/xis/.pyenv/versions/bsb/bin/python[0x50a48d]
[etna:07475] [24] /home/xis/.pyenv/versions/bsb/bin/python(_PyEval_EvalFrameDefault+0x444)[0x50bfb4]
[etna:07475] [25] /home/xis/.pyenv/versions/bsb/bin/python[0x507d64]
[etna:07475] [26] /home/xis/.pyenv/versions/bsb/bin/python(PyEval_EvalCode+0x23)[0x50ae13]
[etna:07475] [27] /home/xis/.pyenv/versions/bsb/bin/python[0x634c82]
[etna:07475] [28] /home/xis/.pyenv/versions/bsb/bin/python[0x4ad695]
[etna:07475] [29] /home/xis/.pyenv/versions/bsb/bin/python(PyRun_InteractiveLoopFlags+0xd4)[0x4afac3]
[etna:07475] *** End of error message ***
Segmentation fault (core dumped)

```

**Desktop/Environment (please complete the following information):**
 - Ubuntu 16.04
 - Python 3.6.9
 - nest-2.18
 - Installation: with MPI, OpenMP and MUSIC as follows:
```
cmake -Dwith-mpi=ON -Dwith-openmp=ON -Dwith-music=ON -DMUSIC_ROOT_DIR=/home/xis/HBP/MUSIC-1.1.16 -DCMAKE_INSTALL_PREFIX:PATH=/home/xis/HBP/nest-2.18.0 /home/xis/HBP/nest-simulator-2.18.0/
```

**Additional context**
Nest, mpi4py and MUSIC are installed in a pyenv environement. Installation of mpi4py works, and it links to a system-level openmpi installation.

Installation:
```
pip install --global-option=build_ext --global-option="-I/usr/lib/x86_64-linux-gnu/openmpi/include/" --global-option="-L/usr/lib/x86_64-linux-gnu/openmpi/lib" mpi4py
```
Test:
```
$ mpiexec -n 5 python -m mpi4py.bench helloworld
Hello, World! I am process 0 of 5 on etna.
Hello, World! I am process 1 of 5 on etna.
Hello, World! I am process 2 of 5 on etna.
Hello, World! I am process 3 of 5 on etna.
Hello, World! I am process 4 of 5 on etna.
```

**Comment by cristianoalessandro** [comment]
I switched the import statements and I do not get the segmentation fault anymore.

```
$ python
>>>import nest
>>>from mpi4py import MPI
>>>nest.Simulate(10)
```

Whould would nest be imported first? This is a bug in nest that should be resolved!

**Comment by terhorstd** [comment]
Hi @cristianoalessandro,
thanks for reporting this. Somehow this sounds very familiar, it may already be solved in 2.20.0, could you confirm with the newer version?
